docs/samples/Microsoft.ML.Samples/Dynamic/SDCACalibrated.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public class SDCACalibrated_BinaryClassificationExample
+    {
+        /// <summary>
+        /// Call <see cref="SdcaCalibratedBinaryTrainer"/> to train a logistic regression model.
+        /// Training examples are generated by calling <see cref="Samples.Utils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorSamples"/>.
+        /// The type of those examples is <see cref="SamplesUtils.DatasetUtils.FloatLabelFloatFeatureVectorSample"/>,
+        /// which contains one bool label and a float feature vector.
+        /// </summary>
+        public static void SDCACalibrated_BinaryClassification()
+        {
+            // Generate C# objects as training examples.
+            var rawData = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorSamples(100);
+
+            // Information in first example.
+            // Label: true
+            Console.WriteLine("First example's label is {0}", rawData.First().Label);
+            // Features is a 10-element float[]:
+            //   [0]	1.0173254	float
+            //   [1]	0.9680227	float
+            //   [2]	0.7581612	float
+            //   [3]	0.406033158	float
+            //   [4]	0.7588848	float
+            //   [5]	1.10602713	float
+            //   [6]	0.6421779	float
+            //   [7]	1.17754972	float
+            //   [8]	0.473704457	float
+            //   [9]	0.4919063	float
+            Console.WriteLine("First example's feature vector is {0}", rawData.First().Features);
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Step 1: Read the data as an IDataView.
+            var data = mlContext.Data.ReadFromEnumerable(rawData);
+
+            // ML.NET doesn't cache data set by default. Caching is very helpful when working with iterative
+            // algorithms which needs many data passes. Since SDCA is the case, we cache.
```

Comment:
> Since SDCA is the case, we cache. [](start = 55, length = 34)

maybe: "Caching is always recommended when using the StochasticDualCoordinateAscent algorithm"

Reply:
Ok.

---
In reply to: [256258680](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256258680) [](ancestors = 256258680)